packages/contracts-bedrock/VERSIONING.md

```diff
@@ -0,0 +1,128 @@
+  - [Merging Back to Develop After Governance Approval](#merging-back-to-develop-after-governance-approval): Explains how to choose the resulting contract versions when merging back into `develop`.
+- [Changelog](#changelog): A CHANGELOG for contract releases is maintained.
+
+> [!NOTE]
+> The rules described in this document must be enforced manually.
+> Ideally, a check can be added to CI to enforce the conventions defined here, but this is not currently implemented.
+
+## Semver Rules
+
+Version increments follow the [style guide rules](./STYLE_GUIDE.md#versioning) for when to bump major, minor, and patch versions in individual contracts:
+
+> - `patch` releases are to be used only for changes that do NOT modify contract bytecode (such as updating comments).
+> - `minor` releases are to be used for changes that modify bytecode OR changes that expand the contract ABI provided that these changes do NOT break the existing interface.
+> - `major` releases are to be used for changes that break the existing contract interface OR changes that modify the security model of a contract.
+>
+> Bumping the patch version does change the bytecode, so another exception is carved out for this.
+> In other words, changing comments increments the patch version, which changes bytecode. This bytecode
+> change implies a minor version increment is needed, but because it's just a version change, only a
+> patch increment should be used.
+
+## Individual Contract Versioning
+
+Versioning for individual contracts works as follows:
+
+- A contract is only `X.Y.Z` on `develop` if it has been governance approved. If it's `X.Y.Z` before that, it must be on a branch. More on this below.
+- For contracts undergoing development, a `-beta.n` identifier must be appended to the version number.
+- For contracts in a release candidate state, an `-rc.n` identifier must be appended to the version number.
+- For contracts with feature-specific changes, a `+feature-name` identifier must be appended to the version number. See the [Smart Contract Feature Development](https://github.com/ethereum-optimism/design-docs/blob/main/smart-contract-feature-development.md) design document to learn more.
+- When making changes to a contract, always bump to the lowest possible version based on the specific change you are making. We do not want to e.g. optimistically bump to a major version, because protocol development sequencing may change unexpectedly. Use these examples to know how to bump the version:
+  - Example 1: A contract is currently on `1.2.3`.
+    - We don't yet know when the next release of this contract will be. However, you are simply fixing typos in comments so you bump the version to `1.2.4-beta.1`.
+    - The next PR made to that same contract clarifies some comments, so it bumps the version to `1.2.4-beta.2`.
+    - The last PR introduces a breaking change, which bumps the version from `1.2.4-beta.2` to `2.0.0-beta.1`. A `1.2.4-rc.1` and `1.2.4` version both never exist.
```

Comment:
1. I initially interpreted the word 'last' as 'previous'. Perhaps just "The next again". 
2. What is "A `1.2.4-rc.1` and `1.2.4` version both never exist." trying to express?

Reply:
(Using the below suggestion to apply it from the github UI)
```suggestion
    - The next PR introduces a breaking change, which bumps the version from `1.2.4-beta.2` to `2.0.0-beta.1`. A `1.2.4-rc.1` and `1.2.4` version both never exist.
```

> What is "A `1.2.4-rc.1` and `1.2.4` version both never exist." trying to express?

Typically the version increment flow would be beta > rc > production. But in this case it goes from `1.2.4-beta.2` straight to `2.0.0-beta.2`, so the "rc > production" parts were never reached for 1.2.4

Reply:
I don't think you applied it?

Reply:
oops, done now